microcosm_pubsub/dispatcher.py

```diff
@@ -36,19 +39,28 @@ def __init__(self, graph):
         self.send_metrics = graph.pubsub_send_metrics
         self.send_batch_metrics = graph.pubsub_send_batch_metrics
         self.max_processing_attempts = graph.config.sqs_message_dispatcher.message_max_processing_attempts
+        self.loop = asyncio.get_event_loop()
 
     def handle_batch(self, bound_handlers) -> List[MessageHandlingResult]:
         """
         Send a batch of messages to a function.
 
         """
+        executor = concurrent.futures.ThreadPoolExecutor(5)
+
         start_time = time()
 
-        instances = [
-            self.handle_message(message, bound_handlers)
+        instances = []
+
+        threads = [
+            self.loop.run_in_executor(executor, self.handle_message, message, bound_handlers)
             for message in self.sqs_consumer.consume()
         ]
 
+        finished, _ = self.loop.run_until_complete(asyncio.wait(threads, return_when=asyncio.ALL_COMPLETED))
```

Comment:
the most inefficient way to wait for everything to complete.

Reply:
Mind expounding what this means a bit?

Reply:
i'm using asyncio.ALL_COMPLETED so the arrays returned only contain finished tasks. Another way of doing this is to run within a while true to check on status of tasks and process the ones that finish as they finish.